.buildkite/scripts/steps/scalability/benchmarking.sh

```diff
@@ -0,0 +1,122 @@
+
+  cd "$KIBANA_DIR"
+  tar -xzf "${LATEST_RUN_ARTIFACTS_DIR}/kibana-default-plugins.tar.gz"
+  tar -xzf "${LATEST_RUN_ARTIFACTS_DIR}/scalability_traces.tar.gz"
+}
+
+checkout_and_compile_load_runner() {
+  mkdir -p "${KIBANA_LOAD_TESTING_DIR}" && cd "${KIBANA_LOAD_TESTING_DIR}"
+
+  if [[ ! -d .git ]]; then
+    git init
+    git remote add origin https://github.com/elastic/kibana-load-testing.git
+  fi
+  git fetch origin --depth 1 "main"
+  git reset --hard FETCH_HEAD
+
+  KIBANA_LOAD_TESTING_GIT_COMMIT="$(git rev-parse HEAD)"
+  export KIBANA_LOAD_TESTING_GIT_COMMIT
+
+  mvn -q test-compile
+  echo "Set 'GATLING_PROJECT_PATH' env var for ScalabilityTestRunner"
+  export GATLING_PROJECT_PATH="$(pwd)"
+}
+
+upload_test_results() {
+  cd "${KIBANA_DIR}"
+  echo "--- Archive Gatling reports and upload as build artifacts"
+  tar -czf "scalability_test_report.tar.gz" --exclude=simulation.log -C kibana-load-testing/target gatling
+  buildkite-agent artifact upload "scalability_test_report.tar.gz"
+  cd "${LATEST_RUN_ARTIFACTS_DIR}"
+  echo "Upload scalability traces as build artifacts"
+  buildkite-agent artifact upload "scalability_traces.tar.gz"
+}
+
+echo "--- Download the latest artifacts from single user performance pipeline"
+download_artifacts
+
+echo "--- Clone kibana-load-testing repo and compile project"
+checkout_and_compile_load_runner
+
+echo "--- Run Scalability Tests with Elasticsearch started only once and Kibana restart before each journey"
+cd "$KIBANA_DIR"
+node scripts/es snapshot&
+
+esPid=$!
+
+# unset env vars defined in other parts of CI for automatic APM collection of
+# Kibana. We manage APM config in our FTR config and performance service, and
+# APM treats config in the ENV with a very high precedence.
+unset ELASTIC_APM_ENVIRONMENT
+unset ELASTIC_APM_TRANSACTION_SAMPLE_RATE
+unset ELASTIC_APM_SERVER_URL
+unset ELASTIC_APM_SECRET_TOKEN
+unset ELASTIC_APM_ACTIVE
+unset ELASTIC_APM_CONTEXT_PROPAGATION_ONLY
+unset ELASTIC_APM_GLOBAL_LABELS
+unset ELASTIC_APM_MAX_QUEUE_SIZE
+unset ELASTIC_APM_METRICS_INTERVAL
+unset ELASTIC_APM_CAPTURE_SPAN_STACK_TRACES
+unset ELASTIC_APM_BREAKDOWN_METRICS
+
+
+export TEST_ES_DISABLE_STARTUP=true
+ES_HOST="localhost:9200"
+export TEST_ES_URL="http://elastic:changeme@${ES_HOST}"
+# Overriding Gatling default configuration
+export ES_URL="http://${ES_HOST}"
+
+# Pings the ES server every second for 2 mins until its status is green
+curl --retry 120 \
+  --retry-delay 1 \
+  --retry-connrefused \
+  -I -XGET "${TEST_ES_URL}/_cluster/health?wait_for_nodes=>=1&wait_for_status=yellow"
+
+export ELASTIC_APM_ACTIVE=true
+
+for journey in scalability_traces/server/*; do
+    export SCALABILITY_JOURNEY_PATH="$KIBANA_DIR/$journey"
+    echo "--- Run scalability file: $SCALABILITY_JOURNEY_PATH"
+    node scripts/functional_tests \
+      --config x-pack/test/performance/scalability/config.ts \
+      --kibana-install-dir "$KIBANA_BUILD_LOCATION" \
+      --debug
+done
+
+trap "kill -9 $esPid" EXIT
```

Comment:
This will work in the happy path but it does not have the desired effect when things go wrong. You need to put the trap immediately after you capture the PID (line: `esPid=$!`). With the `trap` you just install a handler the code is evaluated at `EXIT`. Therefore you need to place it as early as possible to make sure it is evaluated regardless how the script exits (with or without a failure).